src/hal/lib/hal_lib.c

```diff
@@ -266,8 +266,9 @@ void hal_print_error(const char *fmt, ...)
     va_start(args, fmt);
     const char *prefix = "HAL error: ";
     strncpy(_hal_errmsg, prefix, sizeof(_hal_errmsg));
```

Comment:
This bit is completely wrong too
The size parameter is supposed to be the size of the source parameter (prefix), not the destination parameter (_hal_errmsg)